agent/internal/agent.go

```diff
@@ -168,7 +168,7 @@ func (a *Agent) run(ctx context.Context) error {
 		ContainersReattached: reattached,
 	}}:
 	case <-ctx.Done():
-		return ctx.Err()
+		return fmt.Errorf("context done on sending agent started message: %w", ctx.Err())
```

Comment:
I think we should exclude wrapping `ctx.Err`. with contexts, i feel like you usually only care about the cancellation reason.

Reply:
yeah I went back and fourth with excluding ```ctx.Err```. The thinking was to avoid having a random "context canceled" show up in logs.

Wrapping these won't give the reason so makes sense to me to exclude `context`.

Reply:
feel free to ping me tomorrow and we can chat some. i am waffling a bit, too.